app/AbstractRequest.php

```diff
@@ -341,7 +344,38 @@ protected function responseFromCache(): ?Response
         }
 
         $fromCache = Cache::tags($this->cacheTags)->get($this->cacheKey());
-        return $fromCache ? new Response(...$fromCache) : null;
+        if ($fromCache) {
+            $this->sentLogs = $fromCache['logs'];
+            return new Response(...$fromCache['response']);
+        }
+        return null;
+    }
+
+    /**
+     * Return cache key string based on this class, URL, and request body
+     * @return string
+     */
+    public function cacheKey(): string
```

Comment:
I reorganized the code a little to get all the cache methods together and all the logging methods together.